middle_end/flambda/from_lambda/cps_conversion.ml

```diff
@@ -998,26 +994,62 @@ and cps_switch (switch : proto_switch) ~scrutinee (k : Continuation.t)
             let consts_rev = (arm, cont, None, []) :: consts_rev in
             let wrappers = (cont, action) :: wrappers in
             consts_rev, wrappers)
-      ([], [])
-      switch.consts
+  in
+  let consts_rev, wrappers = convert_arms_rev ([],[]) switch.sw_consts in
+  let blocks_rev, wrappers = convert_arms_rev ([], wrappers)
+    (List.combine block_nums sw_blocks)
   in
   let consts = List.rev consts_rev in
+  let blocks = List.rev blocks_rev in
   let failaction, wrappers =
-    match switch.failaction with
+    match switch.sw_failaction with
     | None -> None, wrappers
     | Some action ->
       let cont = Continuation.create () in
       let action = cps_tail action k k_exn in
       let wrappers = (cont, action) :: wrappers in
       Some (cont, None, []), wrappers
   in
-  let switch : I.switch =
-    { numconsts = switch.numconsts;
+  let const_switch : I.switch =
+    { numconsts = switch.sw_numconsts;
       consts;
       failaction;
     }
   in
+  let block_switch : I.switch =
+    { numconsts = switch.sw_numblocks;
+      consts = blocks;
+      failaction;
+    }
+  in
+  let build_switch scrutinee =
+    let const_switch = I.Switch (scrutinee, const_switch) in
+    let block_switch = cps_non_tail
```

Comment:
Do you think you could generate the `Ilambda` term directly ? It seems unnecessary to go back through `cps_non_tail` for such a trivial expression.